diff -urN viewvc2034org/lib/toutf8.py viewvc2034/lib/toutf8.py
--- viewvc2034org/lib/toutf8.py	1970-01-01 09:00:00.000000000 +0900
+++ viewvc2034/lib/toutf8.py	2008-11-04 13:52:00.362454100 +0900
@@ -0,0 +1,16 @@
+import pykf
+import sys
+def toutf8(txt):
+  """pykf.guess cannot guess utf-8 string (wrong to sjis)"""
+  sys.stderr.write( 'enter toutf8()!!!\r\n')
+  c = pykf.guess(txt)
+  if c is pykf.JIS:
+    u = unicode(txt, 'iso2022_jp')
+    return u.encode('utf-8')
+  for enc in ['utf-8', 'euc_jp', 'shift_jis']:
+    try:
+      u = unicode(txt, enc)
+      return u.encode('utf-8')
+    except UnicodeError:
+      pass
+  return txt
diff -urN viewvc2034org/lib/viewvc.py viewvc2034/lib/viewvc.py
--- viewvc2034org/lib/viewvc.py	2008-10-31 02:37:04.023491000 +0900
+++ viewvc2034/lib/viewvc.py	2008-11-04 21:02:37.302000000 +0900
@@ -37,6 +37,7 @@
 import types
 import urllib
 
+import toutf8
 # These modules come from our library (the stub has set up the path)
 import accept
 import compat
@@ -1318,7 +1319,7 @@
   blame_source = []
   if blame_data:
     for i in blame_data:
-      i.text = cgi.escape(i.text)
+      i.text = cgi.escape(toutf8.toutf8(i.text))
       i.diff_href = None
       if i.prev_rev:
         i.diff_href = request.get_url(view_func=view_diff,
@@ -1360,7 +1361,7 @@
         if not line:
           break
         line_no = line_no + 1
-        item = vclib.Annotation(cgi.escape(line), line_no,
+        item = vclib.Annotation(cgi.escape(toutf8.toutf8(line)), line_no,
                                 None, None, None, None)
         item.diff_href = None
         lines.append(item)
@@ -1403,7 +1404,7 @@
     return None
   if cfg.options.use_localtime:
     localtime = time.localtime(date)
-    return time.asctime(localtime) + ' ' + time.tzname[localtime[8]]
+    return time.asctime(localtime) + ' ' + toutf8.toutf8(time.tzname[localtime[8]])
   else:
     return time.asctime(time.gmtime(date)) + ' UTC'
 
