diff -urN viewvc-1.1.0-beta1/lib/ezt.py viewvc-1.1.0-beta1m/lib/ezt.py
--- viewvc-1.1.0-beta1/lib/ezt.py	2007-04-11 00:26:36.611604000 +0900
+++ viewvc-1.1.0-beta1m/lib/ezt.py	2008-11-13 18:08:34.976283800 +0900
@@ -231,6 +231,7 @@
 #    http://svn.webdav.org/repos/projects/ezt/trunk/
 #
 
+import toutf8
 import string
 import re
 from types import StringType, IntType, FloatType, LongType, TupleType
@@ -790,7 +791,7 @@
   """The format specifier is an unknown value."""
 
 def _raw_printer(ctx, s):
-  ctx.fp.write(s)
+  ctx.fp.write( toutf8.toutf8(s))
   
 def _html_printer(ctx, s):
   ctx.fp.write(cgi.escape(s))
diff -urN viewvc-1.1.0-beta1/lib/toutf8.py viewvc-1.1.0-beta1m/lib/toutf8.py
--- viewvc-1.1.0-beta1/lib/toutf8.py	1970-01-01 09:00:00.000000000 +0900
+++ viewvc-1.1.0-beta1m/lib/toutf8.py	2008-11-13 18:35:34.353283800 +0900
@@ -0,0 +1,14 @@
+import chardet
+
+def toutf8(txt):
+  if not txt:
+    return txt
+  c = chardet.detect( txt)
+  try:
+    u = unicode( txt, c['encoding'])
+    return u.encode('utf-8')
+  except UnicodeError:
+    pass
+  except TypeError:
+    pass
+  return txt
diff -urN viewvc-1.1.0-beta1/lib/viewvc.py viewvc-1.1.0-beta1m/lib/viewvc.py
--- viewvc-1.1.0-beta1/lib/viewvc.py	2008-11-05 06:15:53.520043000 +0900
+++ viewvc-1.1.0-beta1m/lib/viewvc.py	2008-11-13 18:42:29.067283800 +0900
@@ -21,6 +21,7 @@
 debug.t_start('startup')
 debug.t_start('imports')
 
+import toutf8
 # standard modules that we know are in the path or builtin
 import sys
 import os
@@ -1084,14 +1085,21 @@
 def format_log(log, cfg, htmlize=1):
   if not log:
     return log
+  log = toutf8.toutf8( log)
+  length = len( log)
+  i = cfg.options.short_log_len
+  if length > i:
+    if ( ord( log[ i -1]) & 0x80):
+      while (length > i) and (( ord( log[ i]) & 0xc0) == 0x80):
+        i += 1
+    log = log[:i]
   if htmlize:
-    s = htmlify(log[:cfg.options.short_log_len],
-                cfg.options.mangle_email_addresses)
+    s = htmlify(log, cfg.options.mangle_email_addresses)
   else:
-    s = cgi.escape(log[:cfg.options.short_log_len])
+    s = cgi.escape(log)
     if cfg.options.mangle_email_addresses == 2:
       s = re.sub(_re_rewrite_email, r'\1@...', s)
-  if len(log) > cfg.options.short_log_len:
+  if length > i:
     s = s + '...'
   return s
 
@@ -1271,9 +1279,9 @@
                                          escape=1)
   return data
 
-def retry_read(src, reqlen=CHUNK_SIZE):
+def retry_read(src):
   while 1:
-    chunk = src.read(CHUNK_SIZE)
+    chunk = src.readline()
     if not chunk:
       # need to check for eof methods because the cStringIO file objects
       # returned by ccvs don't provide them
@@ -1282,11 +1290,13 @@
         continue
     return chunk
   
-def copy_stream(src, dst, cfg, htmlize=0):
+def copy_stream(src, dst, cfg, htmlize=0, conv=False):
   while 1:
     chunk = retry_read(src)
     if not chunk:
       break
+    if conv:
+      chunk = toutf8.toutf8(chunk)
     if htmlize:
       chunk = htmlify(chunk, mangle_email_addrs=0)
     dst.write(chunk)
@@ -1305,7 +1315,7 @@
   def __call__(self, ctx):
     if self.pretext:
       ctx.fp.write(self.pretext)
-    copy_stream(self.fp, ctx.fp, self.cfg, self.htmlize)
+    copy_stream(self.fp, ctx.fp, self.cfg, self.htmlize,conv=True)
     self.fp.close()
     if self.posttext:
       ctx.fp.write(self.posttext)
@@ -1337,10 +1347,10 @@
                                 get_lexer_for_mimetype, \
                                 get_lexer_for_filename
     try:
-      lexer = get_lexer_for_mimetype(mime_type)
+      lexer = get_lexer_for_mimetype(mime_type, encoding='chardet')
     except ClassNotFound:
       try:
-        lexer = get_lexer_for_filename(filename)
+        lexer = get_lexer_for_filename(filename, encoding='chardet')
       except ClassNotFound:
         use_pygments = 0
   except ImportError:
@@ -2556,7 +2566,7 @@
       line = self.save_line
       self.save_line = None
     else:
-      line = self.fp.readline()
+      line = toutf8.toutf8(self.fp.readline())
 
     if not line:
       if self.state == 'no-changes':
